llava/llava.py

```diff
@@ -69,22 +69,27 @@ def get_input_embeddings(
         if pixel_values is None:
             return self.language_model(input_ids)
 
+        # Get the input embeddings from the language model
         inputs_embeds = self.language_model.model.embed_tokens(input_ids)
+        # get the ouptut hidden states from the vision model
         _, _, hidden_states = self.vision_tower(
             pixel_values.transpose(0, 2, 3, 1), output_hidden_states=True
         )
+        # select the hidden states from the desired layer
         selected_image_feature = hidden_states[self.vision_feature_layer]
 
         if self.vision_feature_select_strategy == "default":
+            # default strategy is to select all the hidden states except the first one (CLS token?)
```

Comment:
The first one is the embedding -> https://github.com/ml-explore/mlx-examples/blob/main/clip/model.py#L280